cblox/include/cblox/core/submap_collection_inl.h

```diff
@@ -340,6 +351,9 @@ void SubmapCollection<SubmapType>::fuseSubmapPair(
     const Transformation& T_G_S2 = submap_ptr_2->getPose();
     const Transformation T_S1_S2 = T_G_S1.inverse() * T_G_S2;
     // Merging the submap layers
+
+    std::unique_lock<std::mutex> lock_submap_1(submap_ptr_1->submap_mutex);
+    std::unique_lock<std::mutex> lock_submap_2(submap_ptr_2->submap_mutex);
```

Comment:
From what I can see, <submap_mutex> is never defined anywhere... Looking deeper I realized this function is never actually called anywhere and therefore never instantiated. That's why it compiles despite using a non-existing member.

Despite the fact it's not used, I would leave it in. For now let's just remove the locks, and maybe add a note somewhere that the function is not threadsafe.

Reply:
The other option is to add mutex to the submap class. I guess that would be useful if one would want to speed up the getProjectedMap function by using this method and parallelizing it. Wdyt?

Reply:
I agree having a mutex here makes sense somehow. I see that you've now added it. The idea seems reasonable to me.

The issue as it is right now is that: because submap_mutex submap mutex is private, it cannot be accessed directly from another class like this. There must be a pattern for this kind of problem but I'm not familiar enough with multithreaded stuff to know it. I could think of a member function returning a unique_lock?

Any ideas @victorreijgwart

Reply:
Having taken a closer look at this I would prefer to remove the locks. Currently, the only function that actually uses them is never instantiated, hence why coding errors have slipped through a couple of times. In light of that, I would prefer to add the locks when they're actually needed and can be tested.

Reply:
If we intend to keep these locks, it'd be important to carefully think them through. Given that they easily create (very nasty) bugs. So unless we really need them, I'd be in favor of removing them for now. We can then carefully reintroduce them later when they actually get used and tested.